nanocurrency/nano-node
# Avoid locking the work_watcher mutex in the request_confirm loop

- Move core_test work_watcher tests to a new file
- Add work_watcher::list_watched() returning a vector of watched roots. A vector is used as the expected size of the list is minimal
- Use new list_watched in the request_confirm loop

When saturated, elections are removed in bulk, so this avoids re-locking the watcher mutex in a tight loop.
